src/layouts/enum.py

```diff
@@ -0,0 +1,43 @@
+from typing import Dict
+
+
+class Enum:
+    """Mapping of str values to uint32 integer."""
+
+    _MAX_VAL = 4294967295  # 2^32 - 1
+
+    def __init__(self, values: Dict[str, int]):
+        self._values: Dict[str, int] = {}
+        prev = -1
+        for key, val in values.items():
+            if val != prev + 1:
+                raise ValueError("invlaid enum sequence", prev, val)
+            if val > self._MAX_VAL:
+                raise ValueError("invalid enum value", val)
+            self._values[key] = val
+            prev = val
+
+    def __validate_key(self, key: str):
+        if key not in self._values:
+            raise ValueError("invalid enum key", key)
+
+    def __find_key(self, value: int) -> str:
+        enum_tuple = next(((k, v) for k, v in self._values.items() if v == value), None)
+        if not enum_tuple:
+            raise ValueError("invalid enum value", value)
+        return enum_tuple[0]
+
+    def key(self, value: int) -> str:
+        return self.__find_key(value)
+
+    def value(self, key: str) -> int:
+        self.__validate_key(key)
+        return self._values[key]
+
+    def decode(self, encoded_enum: bytes) -> str:
+        value = int.from_bytes(encoded_enum, byteorder="little")
```

Comment:
Maybe it will be nice to have the `byteorder` as an argument of `decode` and provide a default value?